packages/host/app/resources/search.ts

```diff
@@ -1,79 +1,189 @@
+import { registerDestructor } from '@ember/destroyable';
+
 import { service } from '@ember/service';
 import { waitForPromise } from '@ember/test-waiters';
-import { tracked } from '@glimmer/tracking';
+import { tracked, cached } from '@glimmer/tracking';
 
 import { restartableTask } from 'ember-concurrency';
 import { Resource } from 'ember-resources';
 import flatMap from 'lodash/flatMap';
 
-import { type RealmCards } from '@cardstack/runtime-common';
+import { stringify } from 'qs';
+import { TrackedMap } from 'tracked-built-ins';
+
+import {
+  subscribeToRealm,
+  isCardCollectionDocument,
+  SingleCardDocument,
+} from '@cardstack/runtime-common';
 
 import type { Query } from '@cardstack/runtime-common/query';
 
 import type { CardDef } from 'https://cardstack.com/base/card-api';
 
+import { type CardResource, getCard, asURL } from './card-resource';
+
 import type CardService from '../services/card-service';
 import type RealmServerService from '../services/realm-server';
 
 interface Args {
   named: {
     query: Query;
     realms?: string[];
+    isLive?: true;
     doWhileRefreshing?: (ready: Promise<void> | undefined) => Promise<void>;
   };
 }
 
 export class Search extends Resource<Args> {
   @service private declare cardService: CardService;
   @service private declare realmServer: RealmServerService;
-  @tracked private _instances: CardDef[] = [];
-  @tracked private _instancesByRealm: RealmCards[] = [];
-  @tracked private staleInstances: CardDef[] = [];
-  @tracked private staleInstancesByRealm: RealmCards[] = [];
   @tracked private realmsToSearch: string[] = [];
   loaded: Promise<void> | undefined;
+  private subscriptions: { url: string; unsubscribe: () => void }[] = [];
+  private cardResources = new Map<string, CardResource>();
+  private seenCardResource = new TrackedMap<string, CardResource>();
+  private currentResults = new TrackedMap<string, CardResource>();
 
   modify(_positional: never[], named: Args['named']) {
-    let { query, realms } = named;
+    let { query, realms, isLive, doWhileRefreshing } = named;
     this.realmsToSearch = realms ?? this.realmServer.availableRealmURLs;
 
     this.loaded = this.search.perform(query);
     waitForPromise(this.loaded);
+
+    if (isLive) {
+      this.subscriptions = this.realmsToSearch.map((realm) => ({
+        url: `${realm}_message`,
+        unsubscribe: subscribeToRealm(
+          `${realm}_message`,
```

Comment:
This is a bug. We `subscribeToRealm` to the `${realm}` not `${realm}_message` to get the auth token from local storage.

Reply:
```suggestion
          realm,
```

Reply:
thanks!